include/path_tracking_pid/details/second_order_lowpass.hpp

```diff
@@ -4,26 +4,41 @@
 
 namespace path_tracking_pid::details
 {
-// Error tracker for the last 3 error and filtered error values.
+/**
+ * @brief Discrete time second order lowpass filter
+ */
 class SecondOrderLowpass
 {
 public:
-  // Pushes the given value to the errors FIFO buffer. A corresponding filtered error value is calculated and pushed
-  // to the filtered errors FIFO buffer.
-  void push(double value);
+  /**
+   * @brief Construct a SecondOrderLowpass instance with NaNs
+   */
+  SecondOrderLowpass();
 
-  // Resets both errors and filtered errors FIFO buffers.
-  void reset();
+  /**
+   * @brief Construct a SecondOrderLowpass instance
+   * @param cutoff frequency in Hz, 0 disables the filter
+   * @param damping frequency in Hz
+   */
+  SecondOrderLowpass(double cutoff, double damping);
+
+  void configure(double cutoff, double damping);
 
-  // Read-only access to the errors FIFO buffer.
-  const FifoArray<double, 3> & errors() const;
+  /**
+   * @brief Filter one sample of a signal
+   * @param u signal to be filtered
+   * @param step_size
+   * @return lowpass-filtered signal
+   */
+  double filter(double u, double step_size);
 
-  // Read-only access to the filtered errors FIFO buffer.
-  const FifoArray<double, 3> & filtered_errors() const;
+  void reset();
```

Comment:
Not all member functions have been documented. Intended? Or did you forget?

Reply:
Good point, I'll do that

Reply:
I've updated the docstrings